apis/config/v1alpha2/configuration_types.go

```diff
@@ -27,6 +27,10 @@ import (
 type Configuration struct {
 	metav1.TypeMeta `json:",inline"`
 
+	// Namespace is the namespace in which kueue is deployed. It is used as part of DNSName of the webhook Service.
+	// Defaults to kueue-system.
+	Namespace *string `json:"namespace,omitempty"`
```

Comment:
I think we can avoid the pointers.

If the field is empty `len(field)==0`, you can set the default.

Reply:
As described in [this](https://github.com/kubernetes-sigs/kueue/pull/362#discussion_r959236593) by @kerthcet, I changed *string since this field is optional. Should I change it to string?

> Since this is optional, maybe *string will be better.

Reply:
Still think pointer is a better choice since this is optional, if we set to string, `len(field) == 0` means both unset or misconfiguration of empty string. We can't tell the difference.

Reply:
Do we have any obviously disadvantages by using pointer?

Reply:
@alculquicondor

Reply:
empty string is not a valid value, so we can override it during defaulting. Using pointer just adds extra code that we can avoid. But I'm ok with it, if you feel strongly about it.

Reply:
I agree with @kerthcet. So, I'd like to use `*string`.
Do you have any concerns? @kerthcet

Reply:
sg, please squash.

Reply:
I have squashed.